__tests__/routes/studentRoutes.test.js

```diff
@@ -525,6 +525,106 @@ describe('Update points and level', () => {
   });
 });
 
+describe('Routes for leaderboard', () => {
+  beforeEach(async () => {
+    fakeStudent.completedCourses = [
+      {
+        courseId: fakeCourse._id,
+        totalPoints: 0,
+        isComplete: true,
+        completionDate: new Date(),
+        completedSections: [
+          {
+            sectionId: fakeSection._id,
+            totalPoints: 0,
+            isComplete: true,
+            completionDate: new Date(),
+            completedExercises: [
+              {
+                exerciseId: fakeExercise._id,
+                isComplete: true,
+                completionDate: new Date(),
+                pointsGiven: 10
+              }
+            ]
+          }
+        ]
+      }
+    ]
+
+    // Create a baseUser as the leaderboard also fetches the first- and last name of the user
+    const fakeUser = {
+      _id: mongoose.Types.ObjectId(fakeStudent.baseUser),
+      firstName: 'John',
+      lastName: 'Doe',
+    };
+
+    await db.collection('students').updateOne({ baseUser: fakeStudent.baseUser }, { $set: fakeStudent });
+    await db.collection('users').insertOne(fakeUser)
+  });
+
+  afterEach(async () => {
+    // Remove the user from the database after each test
+    await db.collection('students').deleteOne({ baseUser: userId });
+    await db.collection('users').deleteOne({ _id: userId });
+  });
+
+  it('Gets the leaderboard for all time', async () => {
+    const response = await request(`http://localhost:${PORT}`)
+      .get('/api/students/leaderboard')
+      .send({
+        timeInterval: 'all'
+      })
+      .expect(200);
+
+    expect(response.body).toBeInstanceOf(Array);  
```

Comment:
The tests only check if the response is an array. Maybe it would be nice to check if the array contains at least one of the correct objects 😄